app/ports/models/port.py

```diff
@@ -107,32 +113,31 @@ def load_ports_and_maintainers_table(ports):
                 except KeyError:
                     pass
 
+        @transaction.atomic
         def load_dependencies_table(ports):
-
-            def load_depends(list_of_dependencies, type_of_dependency, port):
+            def load_depends(list_of_dependencies, type_of_dependency, pid):
```

Comment:
Make this `port_id` as well. I would probably reverse the arguments in this function to make it somewhat more "logical": first the port id of the port that you are "modifying", then type of dependencies, then the full list of dependencies.

Reply:
Done, thanks.